bili/tools/api_open_weather.py

```diff
@@ -172,13 +204,17 @@ def execute_query(query: str) -> str:
              if the data could not be obtained.
     :rtype: str
     """
-    # Split the query into city and state if a comma or space is present
-    if "," in query:
-        city, state = map(str.strip, query.split(",", 1))
-    elif " " in query:
-        city, state = map(str.strip, query.split(" ", 1))
+    # Detect if the query is a ZIP code (5-digit numeric)
+    if query.isdigit() and len(query) == 5:
+        lat, lon = get_geocode_from_zip(query)
     else:
-        city, state = query, "CO"  # Default state to "CO" if not provided
+        # Assume it's a city/state input, split the query into city and state if a comma or space is present
+        if "," in query:
+            city, state = map(str.strip, query.split(",", 1))
+        elif " " in query:
+            city, state = map(str.strip, query.split(" ", 1))
+        else:
+            city, state = query, "CO"  # Default to "CO" if no state is provided
 
     # Sanitize and encode the city and state
```

Comment:
Lines 219–224 must be nested inside the outer else block; currently, they override ZIP-based coordinates, breaking ZIP queries and causing incorrect behavior.